ClientRuntimes/Ruby/ms-rest-azure/lib/ms_rest_azure/azure_application_credentials.rb

```diff
@@ -0,0 +1,93 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+
+module MsRestAzure
+  #
+  # Class which represents an Azure token credentials.
+  #
+  class AzureApplicationCredentials < MsRest::ServiceClientCredentials
+
+    TOKEN_ACQUIRE_URL = 'https://login.windows.net/{tenant_id}/oauth2/token'
+    REQUEST_BODY_PATTERN = 'resource={resource_uri}&client_id={client_id}&client_secret={client_secret}&grant_type=client_credentials'
+    RESOURCE_URI = 'https://management.core.windows.net/'
```

Comment:
the token audience should also be settable - both graph.windows.net and keyvault require a different audience, as do azure china, azure stack, etc.